lib/spack/spack/cmd/buildcache.py

```diff
@@ -83,6 +83,9 @@ def setup_parser(subparser):
                               ' its dependencies. Alternatively, one can'
                               ' decide to build a cache for only the package'
                               ' or only the dependencies'))
+    create.add_argument('--public', action='store_true',
+                        help="if this is a public mirror, avoid adding"
+                             " packages when licensing prohibits it")
```

Comment:
I think we should instead default to `public` and allow a flag for `--private`.

I know that *most* build caches are likely going to be private but I think you should have to be explicit to indicate that you're only making this for internal purposes, and that you're ok with distributing otherwise un-distributable packages as long as it's internal.

Making `--public` the special case seems like an easy way for people to distribute things inadvertently.